docs/tutorials/transaction-tutorial.md

```diff
@@ -7,32 +7,37 @@ description: Learn how to perform database operations
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-## Transaction Functions
-
-To perform operations on your application database in DBOS, you use _transaction functions_.
-As their name implies, these functions execute as [database transactions](https://en.wikipedia.org/wiki/Database_transaction).
+Use _transaction functions_ to read and write from your database. A transaction function may contian multiple queries, punctuated by local logic, and executes as a single [database transaction](https://en.wikipedia.org/wiki/Database_transaction). 
 
 Transaction functions must be annotated with the [`@Transaction`](../api-reference/decorators#transaction) decorator and must have a [`TransactionContext`](../api-reference/contexts#transactioncontextt) as their first argument.
 As with other DBOS functions, inputs and outputs must be serializable to JSON.
 
-The [`TransactionContext`](../api-reference/contexts#transactioncontextt) provides a `.client` field you can use to transactionally interact with the database, so you don't need to worry about managing database connections.
+The [`TransactionContext`](../api-reference/contexts#transactioncontextt) provides a `.client` field you can use to interact with the database, so you don't need to worry about managing connections.
 DBOS supports [Knex.js](./using-knex.md), [Drizzle](./using-drizzle.md), [TypeORM](./using-typeorm.md), and [Prisma](./using-prisma.md) clients as well as raw SQL.
 You can configure which client to use in your [`dbos-config.yaml`](../api-reference/configuration.md) file.
-Knex is used by default.
+Knex is the default and we recommend using `Knex.raw()` for raw SQL.
 
-Here's an example of a transaction function:
+See below for an example of a write and a read transaction with each option. Note we recommend marking reads as `{readOnly: true}` for faster performance.
 
 <Tabs groupId="database-clients">
 <TabItem value="knex" label="Knex">
 
 ```javascript
+interface GreetingRecord {
+  name: string;
+  note: string;
+}
+
 export class Greetings {
+  //...
   @Transaction()
-  static async insertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
-    await ctxt.client('greetings').insert({
-      name: friend,
-      note: note
-    });
+  static async insertGreeting(ctxt: TransactionContext<Knex>, gr: GreetingRecord) {
+    await ctxt.client('greetings').insert(gr);
```

Comment:
If it's not too much trouble, might be nice to annotate each RO function with its return type so it's clearer what they're actually doing.

Reply:
done